src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,97 @@
-     */
+    
     public LocalTime addSeconds(LocalTime localTime, Integer secondsToAdd) {
-        return LocalTime.now();
+        return localTime.plusSeconds(secondsToAdd);
     }
-
-    /**
-     * Given the date and the number of weeks to add, return the changed date.
-     */
+    
     public LocalDate addWeeks(LocalDate localDate, Integer numberOfWeeks) {
-        return LocalDate.now();
+        return localDate.plusWeeks(numberOfWeeks);
     }
-
-    /**
-     * Given a random `someDate` date, return one of the following Strings:
-     *     - "`someDate` is after `currentDate`"
-     *                  if `someDate` is in the future relating to the `current date`;
-     *     - "`someDate` is before `currentDate`"
-     *                  if `someDate` is in the past relating to the `current date`;
-     *     - "`someDate` is today"
-     *                  if `someDate` is today;
-     */
+    
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        if (someDate.isAfter(LocalDate.now())) {
+            return someDate.toString() + " is after " + LocalDate.now().toString();
+        }
+        if (someDate.isBefore(LocalDate.now())) {
+            return someDate.toString() + " is before " + LocalDate.now().toString();
+        }
+        return someDate.toString() + " is today";
     }
-
-    /**
-     * Given a String representation of a date and some timezone,
-     * return LocalDateTime in this timezone.
-     */
+    
     public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone) {
-        return LocalDateTime.now();
+        return ZonedDateTime.parse(dateInString)
+                .withZoneSameInstant(ZoneId.of(zone)).toLocalDateTime();
     }
-
-    /**
-     * Given some LocalDateTime, return an OffsetDateTime with the local time offset applied
-     * (`+02:00` for Ukraine).
-     *
-     * Example: we receive a LocalDateTime with a value `2019-09-06T13:17`.
-     *          We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
-     *          where `+02:00` is the offset for our local timezone.
-     *
-     * OffsetDateTime is recommended to use for storing date values in a database.
-     */
+    
     public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
-        return OffsetDateTime.now();
+        return localTime.atOffset(ZonedDateTime.now().getOffset());
     }
-
-    /**
-     * Given a String formatted as `yyyymmdd`,
-     * return Optional of this date as a LocalDate.
-     */
+    
     public Optional<LocalDate> parseDate(String date) {
-        return Optional.empty();
+        if (Integer.parseInt(date.substring(6)) > 31
+                || Integer.parseInt(date.substring(4, 6)) > 12) {
+            return Optional.empty();
+        }
+        return Optional.of(LocalDate.parse(date, DateTimeFormatter.ofPattern("yyyyMMdd")));
```

Comment:
DateTimeFormatter has already constant that you may use - basic iso date
better to use existing constant